### ISSUE TITLE
Usb fedora stuck during boot

### ISSUE THREAD

**Issue by ObelixLeGaulois** [leifliddy/asahi-fedora-usb#33]
Hi, this was working last week, but it seems that now it does not. When booting the usb key, the process stops and the system get stuck on the last line of the screenshot attached.
Then after 20secondes the system make an hard-reset.
Not sure what is happening ....
![Untitled](https://github.com/user-attachments/assets/7babaafc-81b2-4f37-a2b0-5e45d2d2ff8b)

**Comment by a-plastic-bag** [comment]
Reproduced here on my 2021 M1 Pro MacBook Pro. Live environments seem to be my only way to restore my btrfs backups, so without this I can’t fix my Asahi install :(

**Comment by ImanolBarba** [comment]
Same issue :(

**Comment by ImanolBarba** [comment]
I was able to boot by telling grub to load the kernel and initramfs from the working asahi install on the disk (not the one on USB), so either the kernel cmdline or something related to kernel/initramfs installed in the USB is wrong.

**Comment by danacr** [comment]
Same here @ImanolBarba can you share the commands that got it working?

**Comment by ImanolBarba** [comment]
1. Boot with the USB drive
2. Once in GRUB2, input
```
set root=(hdX, Y) 
linux /vmlinuz-ABC ro root=/dev/sdZ verbose
initrd /initramfs-ABC
boot
```

(I'm AFK and I wrote this from memory, I think it should work but I might be forgetting something) 

Where `X` is the disk where asahi /boot is, Y is the partition. `ls` command will list disks

`Z` Is the expected drive name for the usb, if you have no other storage media should be `sda`

`ABC` is the kernel version and release. GRUB2 supports listing files and tab completion. 

This is a shitty workaround, not a proper solution, but can get you out in a pinch (for me it was the need to resize the asahi partition on disk). Essentially what we're doing here is borrowing the main install kernel and initramfs to boot the Linux system in the usb, this works properly enough because it's the same distro, otherwise you might run into breakage if the kernel config was radically different. 

I don't recommend you attempt this unless you're comfortable with GRUB, there shouldn't be any danger of losing data, but definitely danger of losing sanity

Cheers and good luck

**Comment by ImanolBarba** [comment]
Also if I didn't convince you this is a bad workaround, did I mention you have to do this everytime you wanna boot the USB? The only chance of automating this is to write a `loader.conf` entry that would have to be manually updated everytime you update the kernel on the asahi install on disk.

**Comment by danacr** [comment]
thanks @ImanolBarba ! I needed to boot from USB to encrypt the lvm, but even with your suggestion it hangs on boot :(

**Comment by ImanolBarba** [comment]
Sorry to hear that, hopefully the devs will find the issue soon

**Comment by stapelberg** [comment]
No, but the last time I booted a live Asahi was about a year ago, almost to the day :) 

(Nowadays, I just boot the installed version, haven’t had a need to boot a live version.)

**Comment by PatriceBlin** [comment]
For what it is worth, I had some issues as well on a macmini M2. Sharing them here for others.

On the macmini, there is 2 USB-A and 2 USB-C. Both USB-A are not working [because Asahi cannot redistribute the firmwares](https://asahilinux.org/docs/sw/u-boot/).  
Keyboard and USB stick are plugged on USB-C ports. (to interrupt u-boot/grub and send command)

Linux was not able to find the UUID of the root partition from the USB.  
The kernel can be loaded from the USB but not the root partition.  

In my case my Asahi on internal NVME was still working and I noticed that the USB Stick was not visible from the USB-C port but was visible from the USB-A port ... that's why once Linux was booted it could not find the root partition from the USB stick

I followed @ImanolBarba suggestion and used the kernel in internal SSD to boot.
Added a Grub entry (on the NVME)  to boot the internal kernel and used the root UUID from the USB stick and it worked.
Fedora Remix 43 booted from the USB.
